paddle/fluid/framework/details/parallel_ssa_graph_executor.cc

```diff
@@ -13,11 +13,70 @@
 // limitations under the License.
 
 #include "paddle/fluid/framework/details/parallel_ssa_graph_executor.h"
+#include "paddle/fluid/framework/ir/graph_helper.h"
 
 namespace paddle {
 namespace framework {
 namespace details {
 
+std::vector<std::unique_ptr<ir::Graph>> SeparateMultiDevicesGraph(
```

Comment:
this function is not safe. Copying graph can drop graph attrs.
Maybe we shouldn't copy graph. Instead, we pass selected graph ops and vars to threaded graph executor. (change threaded_graph_executor interface.)